File: 卡码/代码随想录/1/04-字符串/Java/src/_02_反转字符串2/LeetCode_0541.java
//541. 反转字符串 II
package _02_反转字符串2;

public class LeetCode_0541 {

}
class Solution {
  public String reverseStr(String s, int k) {
		char[] ch = s.toCharArray();
		for(int i=0;i<ch.length;i+=2*k) {
			int start=i,end=Math.min(start+k-1,ch.length-1);
			while(start<end) {
				ch[start]^=ch[end];
				ch[end]^=ch[start];
				ch[start++]^=ch[end--];
			}
		}
		return new String(ch);
  }
}

File: 卡码/代码随想录/Go/字符串/反转字符串2/LeetCode_0541.go
// 541. 反转字符串 II
package main

import "math"

func reverseStr(s string, k int) string {
	ch:=[]byte(s)
	for i:=0;i<len(ch);i+=2*k {
		start:=i
		end:=int(math.Min(float64(start+k-1),float64(len(ch)-1)))
		for start<end {
			ch[start]^=ch[end]
			ch[end]^=ch[start]
			ch[start]^=ch[end]
			start++
			end--
		}
	}
	return string(ch)
}
